### PR TITLE
Renovation: implement base widget focus and add icon test

### PR DESCRIPTION
**Pull request by LazyLahtak**
1) Implement focus for widget component.
2) Add test for button's icon

### CODE REVIEW

**Review by MaximKudriavtsev — DISMISSED** [review]

File: js/renovation/widget.tsx [hunk 1]
```diff
@@ -269,6 +269,33 @@ export default class Widget {
         return () => active.off(this.widgetRef, { selector, namespace });
     }
 
+    @Effect()
+    focusEffect() {
+        const namespace = 'UIFeedback';
```
Comment (MaximKudriavtsev) [inline js/renovation/widget.tsx]: Same with `namespace`. See [here](https://github.com/DevExpress/DevExtreme/blob/20_1/js/ui/widget/ui.widget.js#L421)

File: testing/jest/button.tests.tsx [hunk 2]
```diff
@@ -116,6 +116,15 @@ describe('Button', () => {
             });
         });
 
+        describe('icon', () => {
+            it('should render icon', () => {
+                const button = render({ icon: 'test' });
+
+                expect(button.is('.dx-button-has-icon')).toBeTruthy();
+                expect(button.exists('.dx-icon.dx-icon-test')).toBeTruthy();
```
Comment (MaximKudriavtsev) [inline testing/jest/button.tests.tsx]: `button.exists('.dx-icon.dx-icon-test')`
->
`button.find('.dx-icon.dx-icon-test').exists()`

**Review by MaximKudriavtsev — DISMISSED** [review]

File: testing/jest/button.tests.tsx (hunk 2 above)
Comment (churkin) [reply testing/jest/button.tests.tsx]: why?

Comment (MaximKudriavtsev) [reply testing/jest/button.tests.tsx]: My fail. I thought that `exists` doesn't have any arguments.

**Review by churkin — APPROVED** [review]

File: testing/jest/widget.tests.tsx [hunk 3]
```diff
@@ -285,32 +285,32 @@ describe('Widget', () => {
...
-            //     expect(widget.hasClass('dx-state-focus')).toBe(false);
-            // });
+                emit(EVENT.blur);
+                expect(widget.hasClass('dx-state-focused')).toBe(false);
+            });
 
-            // it('should not change state if disabled', () => {
-            //     const widget = render({ disabled: true });
+            it('should not change state if disabled', () => {
+                const widget = render({ disabled: true });
```
Comment (MaximKudriavtsev) [inline testing/jest/widget.tests.tsx]: Please add `focusStateEnabled: true` also =)

**Review by MaximKudriavtsev — DISMISSED** [review]

**Review by churkin — DISMISSED** [review]

**Review by MaximKudriavtsev — APPROVED** [review]

**Pull request merged**